packages/core/src/integrations/inboundfilters.ts

```diff
@@ -36,198 +35,184 @@ export class InboundFilters implements Integration {
+    denyUrls: [
+      // eslint-disable-next-line deprecation/deprecation
+      ...(intOptions.blacklistUrls || []),
+      ...(intOptions.denyUrls || []),
+      // eslint-disable-next-line deprecation/deprecation
+      ...(clientOptions.blacklistUrls || []),
+      ...(clientOptions.denyUrls || []),
+    ],
+    ignoreErrors: [...(intOptions.ignoreErrors || []), ...(clientOptions.ignoreErrors || []), ...DEFAULT_IGNORE_ERRORS],
+    ignoreInternal: typeof intOptions.ignoreInternal !== 'undefined' ? intOptions.ignoreInternal : true,
+  };
+}
+
+/** JSDoc */
+export function _shouldDropEvent(event: Event, options: Partial<InboundFiltersOptions>): boolean {
+  if (_isSentryError(event, options.ignoreInternal)) {
+    if (isDebugBuild()) {
+      logger.warn(`Event dropped due to being internal Sentry Error.\nEvent: ${getEventDescription(event)}`);
     }
-    if (this._isIgnoredError(event, options)) {
-      if (isDebugBuild()) {
-        logger.warn(
-          `Event dropped due to being matched by \`ignoreErrors\` option.\nEvent: ${getEventDescription(event)}`,
-        );
-      }
-      return true;
+    return true;
+  }
+  if (_isIgnoredError(event, options.ignoreErrors)) {
+    if (isDebugBuild()) {
+      logger.warn(
+        `Event dropped due to being matched by \`ignoreErrors\` option.\nEvent: ${getEventDescription(event)}`,
+      );
     }
-    if (this._isDeniedUrl(event, options)) {
-      if (isDebugBuild()) {
-        logger.warn(
-          `Event dropped due to being matched by \`denyUrls\` option.\nEvent: ${getEventDescription(
-            event,
-          )}.\nUrl: ${this._getEventFilterUrl(event)}`,
-        );
-      }
-      return true;
+    return true;
+  }
+  if (_isDeniedUrl(event, options.denyUrls)) {
+    if (isDebugBuild()) {
+      logger.warn(
+        `Event dropped due to being matched by \`denyUrls\` option.\nEvent: ${getEventDescription(
+          event,
+        )}.\nUrl: ${_getEventFilterUrl(event)}`,
+      );
     }
-    if (!this._isAllowedUrl(event, options)) {
-      if (isDebugBuild()) {
-        logger.warn(
-          `Event dropped due to not being matched by \`allowUrls\` option.\nEvent: ${getEventDescription(
-            event,
-          )}.\nUrl: ${this._getEventFilterUrl(event)}`,
-        );
-      }
-      return true;
+    return true;
+  }
+  if (!_isAllowedUrl(event, options.allowUrls)) {
+    if (isDebugBuild()) {
+      logger.warn(
+        `Event dropped due to not being matched by \`allowUrls\` option.\nEvent: ${getEventDescription(
+          event,
+        )}.\nUrl: ${_getEventFilterUrl(event)}`,
+      );
     }
-    return false;
+    return true;
   }
+  return false;
+}
 
-  /** JSDoc */
-  private _isSentryError(event: Event, options: Partial<InboundFiltersOptions>): boolean {
-    if (!options.ignoreInternal) {
-      return false;
-    }
+/** JSDoc */
+function _isIgnoredError(event: Event, ignoreErrors: Partial<InboundFiltersOptions>['ignoreErrors']): boolean {
```

Comment:
Although this may make that specific line, I like using the alias here, it makes things more consistent. Readers can scroll to the `InboundFiltersOptions` above and see the type exactly. 

The alias makes the intent of the function very clear, and also communicates how exactly this function is being used - which I think is worth the extra search step a reader has to do to find the exact type.

My rationale here applies for the suggestions below.

Reply:
All of the above said - if we are going to keep it as derived types, I don't think we need the `Partial`, do we?

Reply:
You know what, you're right